flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/AllocatedSlot.java

```diff
@@ -41,7 +41,7 @@
  * an AllocatedSlot was allocated to the JobManager as soon as the TaskManager registered at the
  * JobManager. All slots had a default unknown resource profile. 
  */
-class AllocatedSlot implements PhysicalSlot {
+public class AllocatedSlot implements PhysicalSlot {
```

Comment:
@tillrohrmann I need a subclass of `SlotContext` in this pr. IIRC we've had a conversation where you said that `AllocatedSlot` is package private since it should be used in only `SlotPool`.

Now that `AllocatedSlot` has a public constructor, I'd like to know whether we make this change(add `public`) or introduce a `SettableSlotContext` and mark `AllocatedSlot`'s constructor to package-private.

Reply:
For what do you need a subclass of `SlotContext`? There are already some `SlotContext` test implementations. Maybe you could reuse one of these?

Reply:
I'd like to contain physical slot information to create different `LogicalSlot` with same `SlotContext`.

`SlotContext` has 3 instantiatable subclass. `AllocatedSlot`, `SimpleSlotContext` and `DummySlotContext`.

`DummySlotContext` is a private static inner class. I can extract it and extend a bit for this pr.

`SimpleSlotContext` looks like a legacy class. But...based on its implementation, we can rename it to `TesingSlotContext` because it is quite like a `SettableSlotContext` without too many specific logic.

Reply:
I prefer rename `SimpleSlotContext` to `SettableSlotContext` and use it in this pr, as a follow up, also drop `DummySlotContext`. What do you think? Should it be a separated JIRA?

Reply:
I think you could add it as a separate commit to this PR.

Reply:
done.